docs/rules/0192/trademarked-names.md

```diff
@@ -0,0 +1,88 @@
+---
+rule:
+  aip: 192
+  name: [core, '0192', trademarked-names]
+  summary: Trademarked names should be used correctly.
+permalink: /192/trademarked-names
+redirect_from:
+  - /0192/trademarked-names
+---
+
+# Trademarked names
+
+This rule enforces trademarked names in public comments are not abbreviated and
+follow the trademark owner's branding style, as mandated in [AIP-192][].
+
+## Details
+
+This rule looks at each descriptor in each proto file (exempting oneofs and the
+file itself) and complains if it catches a common trademark mistake. It only
+checks against a known problem list, and does not make guesses beyond that.
+
+It currently catches common mistaken variants for:
+
+- App Engine
+- BigQuery
+- Bigtable
+- Bitbucket
+- Cloud Storage
+- Compute Engine
+- Dataflow
+- Dataprep
+- Dialogflow
+- Directory Sync
+- GitHub
+- GitLab
+- G Suite
+- Pub/Sub
+- Service Mesh
+- Stack Overflow
+
+## Examples
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+message Book {
+  string name = 1;
+
+  // A repository containing Markdown files for each chapter of
+  // the book on Github.
+  // (--           ^ Should be GitHub. --)
+  string github_repo = 2;
```

Comment:
Important to note that `github` is correct in code format (not `git_hub`). What's the correct camel-case for this? `gitHubUsername`? or `githubUsername`? (I think the former...)

Reply:
This _only_ checks comments, not code format.

Reply:
OK. I'll create an issue to check that we don't do things like `git_hub_repo`, etc.